packages/flutter/lib/src/widgets/selectable_region.dart

```diff
@@ -830,6 +840,62 @@ class _SelectableRegionState extends State<SelectableRegion> with TextSelectionD
     await Clipboard.setData(ClipboardData(text: data.plainText));
   }
 
+  bool? _adjustingSelectionExtent;
+  bool _determineIsAdjustingSelectionExtend(bool forward) {
+    if (_adjustingSelectionExtent != null) {
+      return _adjustingSelectionExtent!;
+    }
+    final bool isReversed;
+    final SelectionPoint start = _selectionDelegate.value
+        .startSelectionPoint!;
+    final SelectionPoint end = _selectionDelegate.value.endSelectionPoint!;
+    if (start.localPosition.dy > end.localPosition.dy) {
+      isReversed = true;
+    } else if (start.localPosition.dy < end.localPosition.dy) {
+      isReversed = false;
+    } else {
+      isReversed = start.localPosition.dx > end.localPosition.dx;
```

Comment:
I just tested with chrome, regardless of the text direction, forward=true will always move the selection handle that are lower in screen order. forward =false move the one one higher in screen order. so the logic here should sill hold.

Reply:
Ah ok, I think macOS does the same for bidi when moving the caret (but that may not be true for deletion).  Maybe rename this to left/right instead of forward/backward? I think in editable text land I used forward/backward for logical direction. When we unify the API it will be nice to make that consistent.

Reply:
I renamed extent to end. I am using start and end to describe the handles. Do you think this is good enough? If you meant the input parameter `forward` I think that is accurate since it represent arrow right/arrow down.